src/EnergyPlus/MicroturbineElectricGenerator.cc

```diff
@@ -996,193 +842,137 @@ namespace MicroturbineElectricGenerator {
+            SetupOutputVariable(
+                "Generator Exhaust Air Mass Flow Rate", OutputProcessor::Unit::kg_s, this->ExhaustAirMassFlowRate, "System", "Average", this->Name);
 
-            if (MTGenerator(GeneratorNum).AncillaryPower > 0.0) { // Report Ancillary Power if entered by user
-                SetupOutputVariable("Generator Ancillary Electric Power",
-                                    OutputProcessor::Unit::W,
-                                    MTGeneratorReport(GeneratorNum).AncillaryPowerRate,
-                                    "System",
-                                    "Average",
-                                    MTGenerator(GeneratorNum).Name);
-
-                SetupOutputVariable("Generator Ancillary Electric Energy",
-                                    OutputProcessor::Unit::J,
-                                    MTGeneratorReport(GeneratorNum).AncillaryEnergy,
-                                    "System",
-                                    "Sum",
-                                    MTGenerator(GeneratorNum).Name);
-            }
-            //   Report combustion air outlet conditions if exhaust air calculations are active
-            if (MTGenerator(GeneratorNum).ExhAirCalcsActive) {
-                SetupOutputVariable("Generator Exhaust Air Mass Flow Rate",
-                                    OutputProcessor::Unit::kg_s,
-                                    MTGeneratorReport(GeneratorNum).ExhAirMassFlowRate,
-                                    "System",
-                                    "Average",
-                                    MTGenerator(GeneratorNum).Name);
-                SetupOutputVariable("Generator Exhaust Air Temperature",
-                                    OutputProcessor::Unit::C,
-                                    MTGeneratorReport(GeneratorNum).ExhAirTemperature,
-                                    "System",
-                                    "Average",
-                                    MTGenerator(GeneratorNum).Name);
-            }
+            SetupOutputVariable(
+                "Generator Exhaust Air Temperature", OutputProcessor::Unit::C, this->ExhaustAirTemperature, "System", "Average", this->Name);
         }
     }
 
-    // End of Get Input subroutine for the MT Generator Module
-    //******************************************************************************
+    void MTGeneratorSpecs::simulate(const PlantLocation &EP_UNUSED(calledFromLocation),
+                                    bool EP_UNUSED(FirstHVACIteration),
+                                    Real64 &EP_UNUSED(CurLoad),
+                                    bool EP_UNUSED(RunFlag))
+    {
+        // empty function to emulate current behavior as of conversion to using the PlantComponent calling structure.
+        // calls from the plant side... do nothing.
+        // calls from the ElectricPowerServiceManger call the init, calc, and update worker functions
+    }
```

Comment:
This is a key note.  By changing the factory to return a pointer to the struct instead of a pointer to a plant component, it allows other managers to also get a reference to the same object.  Later, this component struct could also inherit some other base class, like ElectricPowerServiceComponent.  This new base class may have some function like: `eps_simulate(...)`, and the same plant approach can be used over there.  In this particular case, I do wonder if we even _need_ to inherit plant component, but we can leave it for now.